src/templates/finops-hub/modules/dataFactory.bicep

```diff
@@ -4743,6 +4746,121 @@ resource pipeline_ExecuteIngestionETL 'Microsoft.DataFactory/factories/pipelines
   }
 }
 
+//------------------------------------------------------------------------------
+// import_gcp_billing_data pipeline
+//------------------------------------------------------------------------------
+resource importGCPBillingDataPipeline 'Microsoft.DataFactory/factories/pipelines@2018-06-01' = {
+  name: 'import_gcp_billing_data'
+  parent: dataFactory
+  properties: {
+    activities: [
+      {
+        name: 'convert gcp csv'
+        type: 'Copy'
+        dependsOn: [
+          {
+            activity: 'Delete Target'
+            dependencyConditions: [
+              'Completed'
+            ]
+          }
+        ]
+        policy: {
+          timeout: '0.12:00:00'
+          retry: 0
+          retryIntervalInSeconds: 30
+          secureOutput: false
+          secureInput: false
+        }
+        typeProperties: {
+          source: {
+            type: 'DelimitedTextSource'
+            storeSettings: {
+              type: 'GoogleCloudStorageReadSettings'
+              recursive: true
+              wildcardFolderPath: gcpBillingWildcardFolderPath // Parameterized value
+              wildcardFileName: '*'
+              enablePartitionDiscovery: false
+            }
+            formatSettings: {
+              type: 'DelimitedTextReadSettings'
+            }
+          }
+          sink: {
+            type: 'ParquetSink'
+            storeSettings: {
+              type: 'AzureBlobFSWriteSettings'
+            }
+            formatSettings: {
+              type: 'ParquetWriteSettings'
+            }
+          }
+          enableStaging: false
+          translator: {
+            type: 'TabularTranslator'
+            mappings: [
+              {
+                source: {
+                  name: 'BillingAccountId'
+                  type: 'String'
+                  physicalType: 'String'
+                }
+                sink: {
+                  name: 'BillingAccountId'
+                  type: 'String'
+                  physicalType: 'String'
+                }
+              }
+              // Add all other mappings here as per your JSON
+            ]
+            typeConversion: true
+            typeConversionSettings: {
+              allowDataTruncation: true
+              treatBooleanAsNumber: false
+            }
+          }
+        }
+        inputs: [
+          {
+            referenceName: 'GCSbillingexportBucket'
```

Comment:
Missing the definition for the dataset and linked service.

We should store the secret in KeyVault.
There's some logic to only deploy KeyVault in the remote hub scenario so that also needs to be updated - likely by adding a checkbox to the UI to optionally deploy KeyVault when deploying the toolkit.

Reply:
FYI - Moved this to a separate googleCloud.bicep file.

Reply:
Signing off on this to go into the features/xcloud branch. I'll work with Graham to get these changes into a new PR before we close this.